### PR TITLE
Signal receiver that creates UserProfile when User created (fixes #494)

### LINKED ISSUE #494: UserProfile not created – causes 500 Internal Server Error when accessing homepage view
<a href="https://github.com/JarUrb"></img></a> **Issue by [JarUrb](https://github.com/JarUrb)**
_2016-11-07T20:24:24Z_
_Originally opened as https://github.com/stxnext-csr/volontulo/issues/581_

----

**Steps to reproduce**
1. run `python manage.py createsuperuser --settings=volontulo_org.settings.dev`
2. enter username, email address, password
3. run `python manage.py runserver --settings=volontulo_org.settings.dev`
4. go to http://127.0.0.1:8000/admin in web browser
5. log in using username and password provided in step 2
6. go to http://127.0.0.1:8000/ in web browser

**Traceback**
```
Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/

Django Version: 1.8.4
Python Version: 3.4.3
Installed Applications:
('django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'bootstrap3',
 'cookielaw',
 'apps.volontulo',
 'debug_toolbar',
 'django_coverage',
 'django_extensions',
 'django_nose')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware')


Traceback:
File "/home/jarurb/python/volontulo/env/lib/python3.4/site-packages/django/core/handlers/base.py" in get_response
  132.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/home/jarurb/python/volontulo/apps/volontulo/views/__init__.py" in homepage
  45.     if logged_as_admin(request):
File "/home/jarurb/python/volontulo/apps/volontulo/views/__init__.py" in logged_as_admin
  34.         UserProfile.objects.get(user=request.user).is_administrator
File "/home/jarurb/python/volontulo/env/lib/python3.4/site-packages/django/db/models/manager.py" in manager_method
  127.                 return getattr(self.get_queryset(), name)(*args, **kwargs)
File "/home/jarurb/python/volontulo/env/lib/python3.4/site-packages/django/db/models/query.py" in get
  334.                 self.model._meta.object_name

Exception Type: DoesNotExist at /
Exception Value: UserProfile matching query does not exist.
```

### PR DESCRIPTION
**Pull request by JarUrb**
__Story / Bug id:__ https://github.com/CodeForPoznan/volontulo/issues/494

__Reference person:__ @JarUrb

__Description:__

When user is created using `manage.py createsuperuser`, user's profile isn't created. It can lead to internal server errors (HTTP 500).

In commit 2bdb371437d121d73739b981f39b1906c6377606 I added test to show that case. That test would fail without changes in commit bb924b50fffe51093e1ae87cfbf84894539d0797. I also had to fix other tests in commit 1e5d85ced5ccf056874d85f98e961fe4f7cd3f73.

I know that some people don't like signals, but without this change we would have to deal with inconsistent data (i.e. user without corresponding UserProfile) or solve the problem another way (e.g. by subclassing Django's User model and overriding `save()` method) or forbid using `manage.py createsuperuser` entirely.

### CODE REVIEW

File: apps/volontulo/tests/models/test_userprofile.py [hunk 1]
```diff
@@ -18,37 +17,38 @@ def setUp(self):
...
 
         # volunteer user
-        self.volunteer_user = UserProfile.objects.create(
-            user=User.objects.create_user(
-                username='volunteer@example.com',
-                email='volunteer@example.com',
-                password='volunteer',
-            ),
-            is_administrator=False,
+        user = User.objects.create_user(
```
Comment (magul) [inline apps/volontulo/tests/models/test_userprofile.py]: @riv2q DDF support had been merged, but I believe we could skip using it in this PR, as it is already raised.

File: apps/volontulo/__init__.py [hunk 2]
```diff
@@ -3,3 +3,5 @@
 """
 .. module:: __init__
 """
+
+default_app_config = 'apps.volontulo.apps.VolontuloAppConfig'
```
Comment (magul) [inline apps/volontulo/__init__.py]: > New applications should avoid default_app_config. Instead they should require the dotted path to the appropriate AppConfig subclass to be configured explicitly in INSTALLED_APPS.

https://docs.djangoproject.com/en/1.10/ref/applications/

File: apps/volontulo/tests/models/test_userprofile.py (hunk 1 above)
Comment (riv2q) [inline apps/volontulo/tests/models/test_userprofile.py]: yup

**Comment by magul** [comment]
I'm closing it, as it's almost 1 year old, with lot of changes requested and multiple conflicts to be resolved.

**Pull request closed without merging**